### PR TITLE
[INLONG-7146][Sort] Adjust dirty data multiple sink archive strategy

### LINKED ISSUE #7146: [Improve][Sort] Adjust dirty data multiple sink archive strategy
### Description
changes multiple sink archive strategies to dynamically parse arguments, and support helper for multiple sink strategies

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yizhou-Yang**
### Prepare a Pull Request

- Fixes #7146

### Motivation

The original doris dirty data catching requires additional fine-tuning on error catching logic.

### Modifications
1) dirty data&helper: manually format the dirty data to avoid formatting problems caused by the competition for  the ${} regex patternreplaceutils and regex parsing for multiple sink.

2) doris dynamic sink: adjusted respcontent to make sure that the dirty data can be archived correctly.

### CODE REVIEW

**Comment by Yizhou-Yang** [comment]
this should be seperated into 3 different bugfix patches and should be further optimized.
will close this pr later and make it into 3 different prs

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/DirtySinkHelper.java [hunk 1]
```diff
@@ -110,22 +114,59 @@ public void invokeMultiple(T dirtyData, DirtyType dirtyType, Throwable e,
...
+
+        try {
+            // for rowdata where identifier is not the first element, append identifier and SEPARATOR before it.
+            if (dirtyData instanceof RowData) {
+                rootNode = jsonDynamicSchemaFormat.deserialize(((RowData) dirtyData).getBinary(0));
+            } else if (dirtyData instanceof JsonNode) {
+                rootNode = (JsonNode) dirtyData;
+            } else if (dirtyData instanceof String) {
+                // parse and remove the added identifier for string cases
+                String rawIdentifier = ((String) dirtyData).split(SEPARATOR)[0];
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/DirtySinkHelper.java:132]: why do it like this?

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java [hunk 2]
```diff
@@ -130,8 +131,16 @@ public synchronized void invoke(DirtyData<T> dirtyData) throws Exception {
     }
 
     private boolean valid() {
-        return (s3Options.getBatchSize() > 0 && size >= s3Options.getBatchSize())
-                || batchBytes >= s3Options.getMaxBatchBytes();
+        // stash dirty data for at least a minute to avoid flushing too fast
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java:134]: In the current archiving logic, archiving can be done according to time interval or throughput, but the processing here is redundant and inelegant

File: inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java [hunk 3]
```diff
@@ -512,27 +524,58 @@ private void handleDirtyData(Object dirtyData, DirtyType dirtyType, Exception e)
...
+        String table = null;
         try {
-            rootNode = jsonDynamicSchemaFormat.deserialize(((RowData) dirtyData).getBinary(0));
+            if (dirtyData instanceof RowData) {
+                rootNode = jsonDynamicSchemaFormat.deserialize(((RowData) dirtyData).getBinary(0));
+            } else if (dirtyData instanceof JsonNode) {
+                rootNode = (JsonNode) dirtyData;
+            } else if (dirtyData instanceof String) {
+                // parse and remove the added identifier for string cases
+                String[] arr = ((String) dirtyData).split("\\.");
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:542]: why do. it like this?

Comment (Yizhou-Yang) [reply inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:542]: for rowdata in multiple sink, the dirty sink needs to know the identifier. So I passed in  "identifier + DIRTY_SEPARATOR + rowData" to be parsed. The multiple sink scenario all lacks the identifier, so this identifier is added for every single call of string to this method.

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/DirtySinkHelper.java (hunk 1 above)
Comment (Yizhou-Yang) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/DirtySinkHelper.java:132]: for rowdata in multiple sink, the dirty sink needs to know the identifier. So I passed in "identifier + DIRTY_SEPARATOR + rowData" to be parsed for those that misses the identifier. The multiple sink scenario all lacks the identifier, so this identifier is added for every single call of string to this method. This one only has db/table, so it is easier to process

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java (hunk 2 above)
Comment (Yizhou-Yang) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java:134]: OK, then don't merge this part for now...

**Comment by Yizhou-Yang** [comment]
this pr is refactored and then merged

**Pull request closed without merging**